plugin/sampling/strategystore/static/strategy_store.go

```diff
@@ -83,35 +87,86 @@ func (h *strategyStore) Close() {
 	h.cancelFunc()
 }
 
-func (h *strategyStore) autoUpdateStrategies(interval time.Duration, filePath string) {
-	lastValue := ""
+func downloadSamplingStrategies(url string) ([]byte, error) {
+	resp, err := http.Get(url)
+	if err != nil {
+		return nil, fmt.Errorf("failed to download sampling strategies: %w", err)
+	}
+
+	defer resp.Body.Close()
+	if resp.StatusCode != http.StatusOK {
+		if resp.StatusCode == http.StatusServiceUnavailable {
+			return []byte("null"), nil
+		}
+		return nil, fmt.Errorf(
+			"receiving %s while downloading strategies file",
+			resp.Status,
+		)
+	}
+
+	buf := new(bytes.Buffer)
+	if _, err = buf.ReadFrom(resp.Body); err != nil {
+		return nil, fmt.Errorf("failed to read sampling strategies from downloaded JSON: %w", err)
+	}
+	return buf.Bytes(), nil
+}
+
+func isURL(str string) bool {
+	u, err := url.Parse(str)
+	return err == nil && u.Scheme != "" && u.Host != ""
+}
+
+func samplingStrategyLoader(strategiesFile string) strategyLoader {
+	return func() ([]byte, error) {
```

Comment:
nit: you only need to test strategiesFile once, and return different functors. Otherwise `strategyLoader` is pointless, you could've just call the function to read stuff
```
if strategiesFile == "" {
    return func() ([]byte, error) {
        return []byte("null"), nil
    }
}
if isURL(strategiesFile) {
    return func() ([]byte, error) {
        return downloadSamplingStrategies(strategiesFile)
    }
}
```

Reply:
Oops, I was being silly. I get this, thanks.